apps/ledger-live-desktop/src/renderer/modals/CreateLocalManifest/index.tsx

```diff
@@ -0,0 +1,382 @@
+
+import { useManifests } from "@ledgerhq/live-common/platform/providers/RemoteLiveAppProvider/index";
+import FormLiveAppArraySelect from "./FormLiveAppArraySelect";
+import { Separator } from "~/renderer/components/Onboarding/Screens/SelectUseCase/Separator";
+import { DEFAULT_FORM, DEFAULT_VALUES } from "./defaultValues";
+
+function createLocalManifest() {
+  return (
+    <Modal
+      width={600}
+      name="MODAL_CREATE_LOCAL_APP"
+      centered
+      render={({ data, onClose }) => <FormLocalManifest data={data} onClose={onClose} />}
+    />
+  );
+}
+
+function FormLocalManifest({
+  onClose,
+}: {
+  data: { manifest?: LiveAppManifest };
+  onClose: () => void;
+}) {
+  const { addLocalManifest } = useLocalLiveAppContext();
+  const [form, setForm] = useState<LiveAppManifest>({ ...DEFAULT_FORM });
+  const [isDapp, setIsDapp] = useState("dapp" in form);
+
+  const completeManifests = useManifests({ visibility: ["complete"] });
+  const { categories } = useCategories(completeManifests);
+  const walletAPICurrencies: WalletAPICurrency[] = useWalletAPICurrencies();
+
+  DEFAULT_VALUES.categories = categories.filter((category: string) => category !== "all");
+  DEFAULT_VALUES.currencies = walletAPICurrencies.map(currency => currency.id);
+
+  const handleSwitchEthDapp = () => {
+    setIsDapp(prevState => !prevState);
+    setForm((prevState: LiveAppManifest) => {
+      !isDapp
+        ? (prevState.dapp = { ...DEFAULT_FORM.dapp } as LiveAppManifestDapp)
+        : delete prevState.dapp;
+      return prevState;
+    });
+  };
+
+  const formIsValid = useMemo(() => LiveAppManifestSchema.safeParse(form).success, [form]);
+
+  const handleChange = useCallback((path: string, value: unknown) => {
+    setForm((prevState: LiveAppManifest) => {
+      return updateObjectAtPath({ ...prevState }, path, value);
+    });
+  }, []);
+
+  const submitHandler = (e: KeyboardEvent) => {
+    e.preventDefault();
+    if (e.detail !== 0) {
+      formIsValid && addLocalManifest(form);
+      onClose();
+    }
+  };
+
+  return (
+    <form style={{ width: "90%", margin: "auto" }}>
+      <ModalBody
+        onClose={onClose}
+        title={<Trans i18nKey={`settings.developer.createLocalAppModal.title.create`} />}
+        render={() => (
+          <>
+            <ScrollArea>
+              <Flex height={"65vh"} rowGap={15} flexDirection={"column"}>
+                <Flex margin={"auto"} width={"max-content"} columnGap={2}>
+                  <Text marginBottom={1} marginLeft={1} ff="Inter|Medium" fontSize={4}>
+                    {`Wallet-API`}
+                  </Text>
+                  <Switch isChecked={isDapp} onChange={handleSwitchEthDapp} />
+                  <Text marginBottom={1} marginLeft={1} ff="Inter|Medium" fontSize={4}>
+                    {`ETH Dapp`}
+                  </Text>
+                </Flex>
+
+                {Object.keys(form)
+                  .map(key => key as keyof LiveAppManifestSchemaType)
```

Comment:
I would memo for this and it looks like the map is not really necessary and only used for typing the return value